vendor-overwrites/colorpicker/colorconverter.js

```diff
@@ -74,16 +107,21 @@ const colorConverter = (() => {
     const first = num[0];
     if (/rgb/i.test(type)) {
       const k = first.endsWith('%') ? 2.55 : 1;
-      const [r, g, b] = num.map(s => parseFloat(s) * k);
-      return {type: 'rgb', r, g, b, a};
+      const [r, g, b] = num.map(s => {
+        const val = nonDigit.test(s.replace('%', '')) ? -1 : parseFloat(s) * k;
+        // Round the values when converting % to byte value or validation fails
+        return k === 1 ? val : Math.round(val);
+      });
+      return validateRGB(r, g, b) && validateAlpha(a) ? {type: 'rgb', r, g, b, a} : null;
     } else {
-      let h = parseFloat(first);
+      let h = nonDigit.test(first.replace('.', '').replace(/(deg|grad|rad|turn)/, '')) ? -1 : parseFloat(first);
       if (first.endsWith('grad')) h *= 360 / 400;
       else if (first.endsWith('rad')) h *= 180 / Math.PI;
       else if (first.endsWith('turn')) h *= 360;
-      const s = parseFloat(num[1]);
-      const l = parseFloat(num[2]);
-      return {type: 'hsl', h, s, l, a};
+      h %= 360;
+      const s = num[1].includes('%') && parseFloat(num[1]);
```

Comment:
```js
if (!num.slice(1, 3).every(validPercentage)) {
  return null;
}
```

Reply:
I ended up adding percentage, hue and number parsers that return `-1` if invalid. Let me know what you think.